lib/dead_code_detector/class_method_wrapper.rb

```diff
@@ -53,12 +53,17 @@ def target_directory?(method_name)
       return true if DeadCodeDetector.config.ignore_paths.nil?
       source_location = klass.method(method_name).source_location&.first
       return false if source_location.nil?
+      return false if source_location == "(eval)"
       source_location !~ DeadCodeDetector.config.ignore_paths
     end
 
     def owned_method?(method_name)
       original_method = klass.method(method_name)
-      klass.singleton_class <= original_method.owner && !(klass.superclass.singleton_class <= original_method.owner)
+      if klass.respond_to?(:superclass)
```

Comment:
Is there a `klass` that does not have the `superclass` method? I thought every class has this method. -> https://ruby-doc.org/core-2.5.0/Class.html#method-i-superclass

Reply:
Happens if `klass` ends up being a Module